Src/AutoFixtureUnitTest/MailAddressGeneratorTests.cs

```diff
@@ -50,11 +50,36 @@ public void CreateWithNonMailAddressRequestReturnsCorrectResult()
         }
 
         [Fact]
-        public void CreateWithMailAddressRequestReturnsCorrectResult()
+        public void CreatesNoSpecimenWhenContextIsNull()
         {
             // Fixture setup
             var request = typeof(MailAddress);
-            var context = new DelegatingSpecimenContext();
+            var sut = new MailAddressGenerator();
+            //Exercise system
+            var result = sut.Create(request, null);
+            // Verify outcome
+            var expectedResult = new NoSpecimen(request);
+            Assert.Equal(expectedResult, result);
+        }
+
+        [Fact]
+        public void CreateWithMailAddressRequestReturnsCorrectResultUsingLocalPartFromContext()
+        {
+            // Fixture setup
+            var request = typeof(MailAddress);
+            var expectedLocalPart = new EmailAddressLocalPart(Guid.NewGuid().ToString());
+            var context = new DelegatingSpecimenContext()
+            {
+                OnResolve = r =>
+               {
+                   if (typeof(EmailAddressLocalPart).Equals(r))

```

Comment:
How do we feel about [Conditional Test Logic](http://xunitpatterns.com/Conditional%20Test%20Logic.html)?